cmd/satellite/main.go

```diff
@@ -229,6 +244,48 @@ func cmdNodeUsage(cmd *cobra.Command, args []string) (err error) {
 	return generateCSV(ctx, start, end, file)
 }
 
+func cmdValueAttribution(cmd *cobra.Command, args []string) (err error) {
+	ctx := process.Ctx(cmd)
+
+	// Parse the UUID
+	partnerID, err := uuid.Parse(args[0])
+	if err != nil {
+		return errs.Combine(errs.New("Invalid Partner ID format. %s", args[0]), err)
+	}
+
+	layout := "2006-01-02"
+	start, err := time.Parse(layout, args[1])
+	if err != nil {
+		return errs.New("Invalid start date format. Please use YYYY-MM-DD")
+	}
+	end, err := time.Parse(layout, args[2])
+	if err != nil {
+		return errs.New("Invalid end date format. Please use YYYY-MM-DD")
+	}
+
+	// Ensure that start date is not after end date
+	if start.After(end) {
+		return errs.New("Invalid time period (%v) - (%v)", start, end)
+	}
+
+	// send output to stdout
+	if partnerAttribtionCfg.Output == "" {
+		return reports.GenerateAttributionCSV(ctx, partnerAttribtionCfg.Database, *partnerID, start, end, os.Stdout)
+	}
+
+	// send output to file
+	file, err := os.Create(partnerAttribtionCfg.Output)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		err = errs.Combine(err, file.Close())
```

Comment:
we probably want to do something with this error. Maybe log it out for the user?

Reply:
Updated